rir/src/compiler/translations/pir_2_rir.cpp

```diff
@@ -854,16 +854,6 @@ size_t Pir2Rir::compileCode(Context& ctx, Code* code) {
             }
             case Tag::MkFunCls: {
                 auto mkfuncls = MkFunCls::Cast(instr);
-
-                auto dt = DispatchTable::unpack(mkfuncls->code);
-
-                if (dt->capacity() > 1 && !dt->available(1)) {
-                    Pir2Rir pir2rir(compiler, mkfuncls->fun, nullptr, dryRun,
-                                    compiler.logger.get(mkfuncls->fun));
-                    auto rirFun = pir2rir.finalize();
-                    if (!dryRun)
-                        dt->put(1, rirFun);
-                }
```

Comment:
Why is this not necessary any more?

Reply:
this is creating a endless loop if we have a function static call itself...

Reply:
So where does this take place now? Or do we just use the unoptimized version?